source/material_model/melt_boukare.cc

```diff
@@ -0,0 +1,1260 @@
+
+          // Compute endmember molar fractions in the composite.
+          std::vector<double> phase_mole_fractions_in_composite(n_endmembers);
+          double solid_molar_mass = 0.0;
+          double melt_molar_mass = 0.0;
+          double melt_molar_volume = 0.0;
+          double solid_molar_volume = 0.0;
+
+          for (unsigned int i=0; i<n_endmembers; ++i)
+            {
+              // TODO: fix this using a phase vector (phase == 'bridgmanite')
+              const double endmember_phase_fraction_in_solid = i<2 ? bridgmanite_molar_fraction_in_solid : (1.0 - bridgmanite_molar_fraction_in_solid);
+
+              if (endmember_states[i] == EndmemberState::solid)
+                {
+                  phase_mole_fractions_in_composite[i] = solid_molar_fraction * endmember_phase_fraction_in_solid;
+                  endmember_mole_fractions_in_composite[i] = phase_mole_fractions_in_composite[i] * endmember_mole_fractions_per_phase[i];
+                  solid_molar_mass += endmember_mole_fractions_in_composite[i] * molar_masses[i];
+                  solid_molar_volume += endmember_mole_fractions_in_composite[i] * endmembers.volumes[i];
+                }
+              else if (endmember_states[i] == EndmemberState::melt)
+                {
+                  phase_mole_fractions_in_composite[i] = melt_molar_fraction;
+                  endmember_mole_fractions_in_composite[i] = phase_mole_fractions_in_composite[i] * endmember_mole_fractions_per_phase[i];
+                  melt_molar_mass += endmember_mole_fractions_in_composite[i] * molar_masses[i];
+                  melt_molar_volume += endmember_mole_fractions_in_composite[i] * endmembers.volumes[i];
+                }
+              else
+                AssertThrow (false, ExcNotImplemented());
+            }
+
+          const double total_molar_mass = melt_molar_mass + solid_molar_mass;
+          const double total_volume = melt_molar_volume + solid_molar_volume;
+
+          out.thermal_expansion_coefficients[q] = 0.0;
+          out.specific_heat[q] = 0.0;
+          out.compressibilities[q] = 0.0;
+
+          // Average the individual endmember properties
+          for (unsigned int i=0; i<n_endmembers; ++i)
+            {
+              out.thermal_expansion_coefficients[q] += (endmember_mole_fractions_in_composite[i] * endmembers.volumes[i] * endmembers.thermal_expansivities[i]) / total_volume;
+              out.specific_heat[q] += endmember_mole_fractions_in_composite[i] * endmembers.heat_capacities[i] / total_molar_mass;
+
+
+              if (endmember_states[i] == EndmemberState::solid && solid_molar_volume > 0)
+                {
+                  out.compressibilities[q] += (endmember_mole_fractions_in_composite[i] * endmembers.volumes[i])
+                                              / (solid_molar_volume * endmembers.bulk_moduli[i]);
+                }
+            }
+
+          if (solid_molar_volume > 0)
```

Comment:
Could you search through the file and replace all `0` that are used for doubles with `0.0`? It just makes the code harder to read if it is not immediately clear what type the variable has.